topology/probes/sriov/sriov.go

```diff
@@ -0,0 +1,126 @@
+// +build linux
+
+package sriov
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/vishvananda/netlink"
+
+	"github.com/skydive-project/skydive/filters"
+	"github.com/skydive-project/skydive/logging"
+	"github.com/skydive-project/skydive/topology"
+	"github.com/skydive-project/skydive/topology/graph"
+)
+
+// Probe is a Sriov probe designed to handle all vfs on server
+type Probe struct {
+	Graph  *graph.Graph
+	Root   *graph.Node
+	Cancel context.CancelFunc // cancel function
+}
+
+// MANAGER is a const to be used as a string to describe who handles sriov nodes
+const MANAGER = "vfio"
+
+// GenerateNameForNodeFromPhysInterfaceNameAndVFID - generates a name for vf based on physical device name and id of of this vf
+func GenerateNameForNodeFromPhysInterfaceNameAndVFID(PhysName string, ID string) string {
+	return fmt.Sprintf("%s-vf-%s", PhysName, ID)
+}
+
+func allZero(s []byte) bool {
+	for _, v := range s {
+		if v != 0 {
+			return false
+		}
+	}
+	return true
+}
+
+func (h *Probe) getAllSriovNodes() []*graph.Node {
+	filter := filters.NewAndFilter(
+		filters.NewTermStringFilter("Manager", MANAGER),
+	)
+	m := graph.NewElementFilter(filter)
+	return h.Graph.GetNodes(m)
+}
+
+func (h *Probe) periodicChecker(ctx context.Context) {
+	for ctx.Err() == nil {
+		linkList, err := netlink.LinkList()
```

Comment:
I would prefer to have it event based not using polling pattern. Can ProceedNewLink not be used ? The handler could be extended to handled deletion too